core/src/miner/mem_pool.rs

```diff
@@ -1538,6 +1552,83 @@ pub mod test {
         TransactionOrder::for_transaction(&item, 0)
     }
 
+    #[test]
+    fn external_transactions_whose_fees_are_under_the_mem_pool_min_fee_are_rejected() {
+        //Setup test_client
+        let test_client = TestBlockChainClient::new();
+
+        // Set the pay transaction minimum fee
+        let fees = MemPoolFees::create_from_options(
+            Some(150),
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+        );
+
+        let db = Arc::new(kvdb_memorydb::create(crate::db::NUM_COLUMNS.unwrap_or(0)));
+        let mut mem_pool = MemPool::with_limits(8192, usize::max_value(), 3, db, fees);
+
+        let fetch_account = |p: &Public| -> AccountDetails {
+            let address = public_to_address(p);
+            let a = test_client.latest_regular_key_owner(&address).unwrap_or(address);
+            AccountDetails {
+                seq: test_client.latest_seq(&a),
+                balance: test_client.latest_balance(&a),
+            }
+        };
+
+        let keypair = Random.generate().unwrap();
+        let address = public_to_address(keypair.public());
+        test_client.set_balance(address, 1_000_000_000_000);
+
+        let no_timelock = TxTimelock {
+            block: None,
+            timestamp: None,
+        };
+
+        let inserted_block_number = 1;
+        let inserted_timestamp = 100;
+        let inputs = vec![
+            MemPoolInput::new(create_signed_pay_with_fee(0, 200, keypair), TxOrigin::External, no_timelock),
+            MemPoolInput::new(create_signed_pay_with_fee(1, 140, keypair), TxOrigin::External, no_timelock),
+            MemPoolInput::new(create_signed_pay_with_fee(1, 160, keypair), TxOrigin::External, no_timelock),
+            MemPoolInput::new(create_signed_pay_with_fee(2, 149, keypair), TxOrigin::External, no_timelock),
```

Comment:
Would you add test cases for `TxOrigin::Local`?

Reply:
I added a unit test.